tests/pkg/k8sclient/cspc_util.go

```diff
@@ -169,3 +170,159 @@ func (client *Client) GetCSPC(cspcName, cspcNamespace string) *cstor.CStorPoolCl
+			return resourceLimitMatches
+		}
+		time.Sleep(3 * time.Second)
+	}
+	return false
+}
+
+// HasResourceLimitOnPoolManagerEventually ...
+func (client *Client) HasResourceLimitOnPoolManagerEventually(
+	cspcName, cspcNamespace string, expectedResource *corev1.ResourceRequirements) bool {
+	for i := 0; i < (maxRetry + 100); i++ {
+		resourceLimitMatches := true
+		pmList := client.GetPoolManagerList(cspcName, cspcNamespace)
+		for _, pm := range pmList.Items {
+			for _, container := range pm.Spec.Template.Spec.Containers {
+				if container.Name == "cstor-pool" {
+					if !reflect.DeepEqual(container.Resources, *expectedResource) {
+						resourceLimitMatches = false
+					}
+				}
+			}
+		}
+		if resourceLimitMatches {
+			return resourceLimitMatches
+		}
+		time.Sleep(3 * time.Second)
+	}
+	return false
+}
+
+// HasTolerationsOnCSPIEventually ...
+func (client *Client) HasTolerationsOnCSPIEventually(
+	cspcName, cspcNamespace string, tolerations []corev1.Toleration) bool {
+	for i := 0; i < (maxRetry + 100); i++ {
+		tolerationsMatches := true
+		cspiList := client.GetCSPIList(cspcName, cspcNamespace)
+		for _, v := range cspiList.Items {
+			if !reflect.DeepEqual(v.Spec.PoolConfig.Tolerations, tolerations) {
+				tolerationsMatches = false
+			}
+		}
+		if tolerationsMatches {
+			return tolerationsMatches
+		}
+		time.Sleep(3 * time.Second)
+	}
+	return false
+}
+
+// HasTolerationsOnPoolManagerEventually ...
+func (client *Client) HasTolerationsOnPoolManagerEventually(
+	cspcName, cspcNamespace string, tolerations []corev1.Toleration) bool {
+	for i := 0; i < (maxRetry + 100); i++ {
+		tolerationsMatches := true
+		pmList := client.GetPoolManagerList(cspcName, cspcNamespace)
+		for _, pm := range pmList.Items {
+			if !reflect.DeepEqual(pm.Spec.Template.Spec.Tolerations, tolerations) {
+				tolerationsMatches = false
+			}
+		}
+		if tolerationsMatches {
+			return tolerationsMatches
+		}
+		time.Sleep(3 * time.Second)
+	}
+	return false
+}
+
+// HasPriorityClassOnCSPIEventually ...
+func (client *Client) HasPriorityClassOnCSPIEventually(
+	cspcName, cspcNamespace string, priorityClass *string) bool {
+	for i := 0; i < (maxRetry + 100); i++ {
+		priorityClassMatches := true
+		cspiList := client.GetCSPIList(cspcName, cspcNamespace)
+		for _, v := range cspiList.Items {
+			if !reflect.DeepEqual(v.Spec.PoolConfig.PriorityClassName, priorityClass) {
+				priorityClassMatches = false
+			}
+		}
+		if priorityClassMatches {
+			return priorityClassMatches
+		}
+		time.Sleep(3 * time.Second)
+	}
+	return false
+}
+
+// HasPriorityClassOnPoolManagerEventually ...
+func (client *Client) HasPriorityClassOnPoolManagerEventually(
+	cspcName, cspcNamespace string, priorityClass *string) bool {
+	for i := 0; i < (maxRetry + 100); i++ {
+		priorityClassMatches := true
+		pmList := client.GetPoolManagerList(cspcName, cspcNamespace)
+		for _, pm := range pmList.Items {
+			if !reflect.DeepEqual(pm.Spec.Template.Spec.PriorityClassName, *priorityClass) {
```

Comment:
Can we make use of `==` operator? instead of reflect.DeeqEqual?

Reply:
Just kept like this for consistency. Are there any concerns?